core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -913,9 +922,11 @@ private void initLMPreparationHandler() {
 
         for (LMProfileConfig lmConfig : lmPreparationHandler.getLMProfileConfigs()) {
             ProfileConfig profile = profilesByName.get(lmConfig.getProfile());
-            FlagEncoder encoder = encodingManager.getEncoder(profile.getVehicle());
-            // note that we do not consider turn costs during LM preparation?
-            Weighting weighting = createWeighting(new HintsMap(profile.getWeighting()), encoder, NO_TURN_COST_PROVIDER);
+            // note that we do not consider turn costs during LM preparation. this is important if we want
+            // to allow e.g. changing the u_turn_costs per request (we have to use the minimum weight settings (
+            // = no turn costs) for the preparation)
+            // todonow: is it too ugly this way?
+            Weighting weighting = createWeighting(profile, new PMap().put("__disable_turn_costs", true));
```

Comment:
Any opinion here? `createWeighting` normally uses a `DefaultTurnCostProvider` if `profile.isTurnCosts()==true`. But for LM we always want to enforce not using turn costs for the preparation, but to me it seemed like this does not justify adding an extra parameter, so I am using the pmap workaround here...

Reply:
I would not do this here and just force the usage of no turn cost provider in LM. IMO this is something implementation specific?

Reply:
What do you mean? You mean we should always set `turn_costs: false` for all profiles with LM? But then we cannot use turn costs with LM...

Reply:
Ah you mean somehow disabling the turn costs in `PrepareLandmarks`? Yes that would be the right thing to do, but how? `PrepareLandmarks` just gets a `Weighting`, but how would it disable the turn costs then?

Reply:
> Ah you mean somehow disabling the turn costs in PrepareLandmarks?

Yes. Isn't it already ignored when we do `traversalMode = TraversalMode.NODE_BASED` in LandmarkStorage?

Reply:
Ah yes thats a good point. Yes that should fix it :+1: